src/store/Controller.cc

```diff
@@ -708,20 +708,29 @@ Store::Controller::updateOnNotModified(StoreEntry *old, StoreEntry &e304)
     //         sake, it is best to detect and skip such repeated update calls.
     if (e304.mem_obj->appliedUpdates) {
         debugs(20, 5, "ignored repeated update of " << *old << " with " << e304);
-        return;
+        return true;
     }
     e304.mem_obj->appliedUpdates = true;
 
-    if (!old->updateOnNotModified(e304)) {
-        debugs(20, 5, "updated nothing in " << *old << " with " << e304);
-        return;
+    try {
+        if (!old->updateOnNotModified(e304)) {
+            debugs(20, 5, "updated nothing in " << *old << " with " << e304);
+            return true;
+        }
+    } catch (...) {
+        debugs(20, DBG_IMPORTANT, "ERROR: Failed to update a cached response: " << CurrentException <<
+               Debug::Extra << "cache entry that needed revalidation: " << *old <<
+               Debug::Extra << "304/revalidation response: " << e304);
```

Comment:
I am not sure we should report these low-level details to cache.log (at level 1). They might be handy in developer-driven triage, but even that assertion is questionable, and admins would not know what to do with this information. If others think that these details should not be dumped, let's remove them!

N.B. A part of StoreEntry gists printed by these two lines is the StoreEntry object memory address. Exposing that info has been flagged as a potential threat in the past.

Reply:
Removed in branch commit f7d15b9.